2_munge/src/aggregate_inputs.R

```diff
@@ -46,105 +46,81 @@ aggregate_nwis <- function(ind_file, raw_ind_file, remake_file, sites_yml, gd_co
-  # this is just a place holder for now until nwm nc files are in correct format
 
-####### commenting out for now until nwm nc issues are sorted out; creating fake hydrographs below #####
   # aggregating NWM data to a daily scale
-  # sites = yaml::yaml.load_file(sc_retrieve(sites_yml,remake_file = remake_file))
-  #
-  # comids_lookup = readr::read_delim(sc_retrieve(comids_file, remake_file = remake_file), delim='\t')
-  #
-  # input_raw = nc_open(sc_retrieve(raw_ind_file, remake_file = remake_file))
-  #
-  # keep <- input_raw$dim$feature_id$vals %in% comids_lookup$COMID # comids
-  #
-  # dimids <- input_raw$var$streamflow$dimids
-  #
-  # site_inds <- which(keep)
-  #
-  # if(length(dimids) == 2) {
-  #   streamflow <- matrix(nrow=input_raw$dim$time$len, ncol=length(site_inds))
-  # } else if (length(dimids) == 3) {
-  #   if(!all(input_raw$var$streamflow$dimids == c(0,2,1))) stop("dimids now as expected")
-  #   streamflow <- array(dim = c(length(site_inds), input_raw$dim$time$len, input_raw$dim$reference_time$len))
-  # }
-  #
-  # for(s in 1:length(site_inds)) {
-  #   if(length(dimids) == 2) {
-  #     # Note axis order is assumed here!!!
-  #     streamflow[,s] <- ncvar_get(input_raw, input_raw$var$streamflow,
-  #                                 start = c(site_inds[s], 1),
-  #                                 count = c(1, -1))
-  #     # time <- input_raw$var$streamflow$dim[[2]]$vals # time value; minutes since 1970-01-01 00:00:00 UTC
-  #
-  #
-  #
-  #   } else if(length(dimids) == 3) {
-  #     for(r in 1:input_raw$dim$reference_time$len) {
-  #       streamflow[s,,r] <- ncvar_get(input_raw, input_raw$var$streamflow,
-  #                                     start = c(site_inds[s], 1, r),
-  #                                     count = c(1, -1, 1))
-  #     }
-  #   }
-  # }
-  # # stream flow is [site, valid time, reference time]
-  #
-  # ncvar_put(new_nc, new_nc$var$streamflow, streamflow)
-  #
-  # if("reference_time" %in% names(nc$dim)) {
-  #   ncvar_put(new_nc, "reference_time", nc$dim$reference_time$vals)
-  # }
-  #
-  #
-######
-
-  # agg_nwm = lapply(sites, function(site){
-  #   agg_out = input_raw$nitrate_sensor %>%
-  #     dplyr::filter(site_no == site) %>% # stats package overrides dplyr filter()
-  #     mutate(date = as.Date(dateTime)) %>%
-  #     group_by(date) %>%
-  #     summarise(X_99133_000000_mean = mean(X_99133_00000)) %>%
-  #     mutate(site_no = site, tz_cd = 'UTC')
-  # }) %>% bind_rows() %>% as_data_frame()
-
-  #fake hydrograph for now
-  date_retro = seq(as.Date('1993-01-01'), as.Date('2018-03-01'), by = 'days') # retro dates
-  date_forecast = seq(as.Date('2017-01-01'), as.Date('2018-05-01'), by = 'days') # reference dates
-  model_range = ifelse(length(grep('retro', ind_file))>0,NA, ifelse(length(grep('med', ind_file))>0, 10, 30)) # model range in days
-
   sites = yaml::yaml.load_file(sc_retrieve(sites_yml,remake_file = remake_file))
+
```

Comment:
A new option for getting the final COMIDs list is to pull site_info$COMID from 1_data/out/site_info.tsv, which is now a target in 2_munge - the site_info file only has the final 3 sites but also has a COMID column, so you don't need to load separate files for the site list and comid lookup.